types/watecco/triphaso/uplink.js

```diff
@@ -0,0 +1,437 @@
+
+        if (cmdID === 0x8a) {
+          decoded.zclheader.alarm = 1;
+        } else {
+          decoded.zclheader.alarm = 0;
+        }
+
+        let index = 0;
+        // data index start
+        if ((cmdID === 0x0a) | (cmdID === 0x8a)) {
+          index = 7;
+        }
+        // if (cmdID === 0x01) {index = 8; decoded.zclheader.status = bytes[6];}
+
+        // on/off present value
+        if ((clusterdID === 0x0006) & (attributeID === 0x0000)) {
+          const state = bytes[index];
+          if (state === 0) {
+            tab[`output${decoded.zclheader.endpoint + 1}`] = "OFF";
+          }
+          if (state === 1) {
+            tab[`output${decoded.zclheader.endpoint + 1}`] = "ON";
+          }
+        }
+
+        // energy and power metering
+        if ((clusterdID === 0x800a) & (attributeID === 0x0000)) {
+          const phase = {};
+
+          phase[`pActiveEnergy${key}`] = UintToInt(
+            bytes[index + 1] * 256 * 256 * 256 +
+              bytes[index + 2] * 256 * 256 +
+              bytes[index + 3] * 256 +
+              bytes[index + 4],
+            4,
+          );
+          phase[`nActiveEnergy${key}`] = UintToInt(
+            bytes[index + 5] * 256 * 256 * 256 +
+              bytes[index + 6] * 256 * 256 +
+              bytes[index + 7] * 256 +
+              bytes[index + 8],
+            4,
+          );
+          phase[`pReactiveEnergy${key}`] = UintToInt(
+            bytes[index + 9] * 256 * 256 * 256 +
+              bytes[index + 10] * 256 * 256 +
+              bytes[index + 11] * 256 +
+              bytes[index + 12],
+            4,
+          );
+          phase[`nReactiveEnergy${key}`] = UintToInt(
+            bytes[index + 13] * 256 * 256 * 256 +
+              bytes[index + 14] * 256 * 256 +
+              bytes[index + 15] * 256 +
+              bytes[index + 16],
+            4,
+          );
+          phase[`pActivePowerW${key}`] = UintToInt(
+            bytes[index + 17] * 256 * 256 * 256 +
+              bytes[index + 18] * 256 * 256 +
+              bytes[index + 19] * 256 +
+              bytes[index + 20],
+            4,
+          );
+          phase[`nActivePowerW${key}`] = UintToInt(
+            bytes[index + 21] * 256 * 256 * 256 +
+              bytes[index + 22] * 256 * 256 +
+              bytes[index + 23] * 256 +
+              bytes[index + 24],
+            4,
+          );
+          phase[`pReactivePowerVar${key}`] = UintToInt(
+            bytes[index + 25] * 256 * 256 * 256 +
+              bytes[index + 26] * 256 * 256 +
+              bytes[index + 27] * 256 +
+              bytes[index + 28],
+            4,
+          );
+          phase[`nReactivePowerVar${key}`] = UintToInt(
+            bytes[index + 29] * 256 * 256 * 256 +
+              bytes[index + 30] * 256 * 256 +
+              bytes[index + 31] * 256 +
+              bytes[index + 32],
+            4,
+          );
+
+          emit("sample", {
+            topic: `phase_${topic}`,
```

Comment:
use more speaking variable names: e.g. phase to describe the phase as a string and phaseData to describe the resulting payload